lib/matplotlib/widgets.py

```diff
@@ -2007,113 +2050,216 @@ def new_axes(self, ax):
-            self.stay_rect.set_x(self.rect.get_x())
-            self.stay_rect.set_y(self.rect.get_y())
-            self.stay_rect.set_width(self.rect.get_width())
-            self.stay_rect.set_height(self.rect.get_height())
-            self.stay_rect.set_visible(True)
+        return False
 
-        self.canvas.draw_idle()
-        vmin = self.pressv
-        xdata, ydata = self._get_data(event)
-        if self.direction == 'horizontal':
-            vmax = xdata or self.prev[0]
-        else:
-            vmax = ydata or self.prev[1]
+    def _release(self, event):
+        """Button release event handler."""
+        if not self.interactive:
+            self.rect.set_visible(False)
 
-        if vmin > vmax:
-            vmin, vmax = vmax, vmin
+        vmin, vmax = self.extents
         span = vmax - vmin
-        if self.minspan is not None and span < self.minspan:
+        if span < self.minspan:
+            self.set_visible(False)
+            self.update()
             return
+
         self.onselect(vmin, vmax)
-        self.pressv = None
+        self.update()
+
         return False
 
     def _onmove(self, event):
         """Motion notify event handler."""
-        if self.pressv is None:
-            return
 
-        self._set_span_xy(event)
+        v = event.xdata if self.direction == 'horizontal' else event.ydata
+        if self.direction == 'horizontal':
+            vpress = self.eventpress.xdata
+        else:
+            vpress = self.eventpress.ydata
 
-        if self.onmove_callback is not None:
-            vmin = self.pressv
-            xdata, ydata = self._get_data(event)
-            if self.direction == 'horizontal':
-                vmax = xdata or self.prev[0]
-            else:
-                vmax = ydata or self.prev[1]
+        # move existing span
+        # When "dragging from anywhere", the `self.active_handle` is set to 'C'
+        # in _set_active_handle (match notation used in the RectangleSelector)
+        if self.active_handle == 'C' and self._extents_on_press is not None:
+            vmin, vmax = self._extents_on_press
+            dv = v - vpress
+            vmin += dv
+            vmax += dv
 
+        # resize an existing shape
+        elif self.active_handle and self.active_handle != 'C':
+            vmin, vmax = self._extents_on_press
+            if self.active_handle == 'min':
+                vmin = v
+            else:
+                vmax = v
+        # new shape
+        else:
+            vmin, vmax = vpress, v
             if vmin > vmax:
                 vmin, vmax = vmax, vmin
+
+        self.extents = vmin, vmax
+
+        if self.onmove_callback is not None:
             self.onmove_callback(vmin, vmax)
 
-        self.update()
         return False
 
-    def _set_span_xy(self, event):
-        """Set the span coordinates."""
-        x, y = self._get_data(event)
-        if x is None:
-            return
+    def draw_shape(self, vmin, vmax):
```

Comment:
should this be private?

Reply:
Yes, I agree that this should be private. I didn't make it private to be consistent with the API of `RectangleSelector` and `EllipseSelector`. Instead, would it be better to make it private for all 3 with a `@_api.deprecated('3.5')` in the case of `RectangleSelector` and `EllipseSelector`?

Reply:
Went ahead and privatise for all 3.